File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        freq = dict()
        ans = 0
        
        for cur in nums:
            if freq.get(k - cur, 0) > 0:
                ans += 1
                freq[k - cur] -= 1
            else:
                freq[cur] = freq.get(cur, 0) + 1
        return ans

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        return solve(nums, k);
    }
private:
    int solve(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        int ans = 0;
        
        for(auto cur: nums){
            if(freq[k - cur] > 0) {
                ans++; 
                freq[k - cur]--;
            }
            else 
                freq[cur]++;
        }
        return ans;
    }
};